wpilibj/src/athena/java/edu/wpi/first/wpilibj/DoubleSolenoid.java

```diff
@@ -82,22 +76,22 @@ public DoubleSolenoid(final int moduleNumber, final int forwardChannel,
       throw ex;
     }
 
-    m_forwardMask = (byte) (1 << m_forwardChannel);
-    m_reverseMask = (byte) (1 << m_reverseChannel);
+    m_forwardMask = (byte) (1 << forwardChannel);
+    m_reverseMask = (byte) (1 << reverseChannel);
 
-    HAL.report(tResourceType.kResourceType_Solenoid, m_forwardChannel,
+    HAL.report(tResourceType.kResourceType_Solenoid, forwardChannel,
                                    m_moduleNumber);
-    HAL.report(tResourceType.kResourceType_Solenoid, m_reverseChannel,
+    HAL.report(tResourceType.kResourceType_Solenoid, reverseChannel,
                                    m_moduleNumber);
-    LiveWindow.addActuator("DoubleSolenoid", m_moduleNumber, m_forwardChannel, this);
+    LiveWindow.addActuator("DoubleSolenoid", m_moduleNumber, forwardChannel, this);
   }
 
   /**
    * Destructor.
    */
   public synchronized void free() {
     SolenoidJNI.freeSolenoidPort(m_forwardHandle);
-    SolenoidJNI.freeSolenoidPort(m_reverseChannel);
+    SolenoidJNI.freeSolenoidPort(m_reverseHandle);
```

Comment:
**This is an implementation change**

Is this what is should be instead of channel?